lib/lf/mesh/utils/write_tikz.h

```diff
@@ -6,26 +6,120 @@
- * @param filename name of output file
  *
- * This function writes a file of code, which included in LaTeX draws your
- * entity using TikZ Graphics.
+ * @param mesh the mesh to be stored to file
+ * @param filename name of output file.
+ * @param output_ctrl enum flags controlling amount of output
+ *
+ * This function writes a file of code, which included in LaTeX draws a
+ * mesh using TikZ Graphics.
+ * In particular, edges, cells and nodes in the mesh can be written to file and
+visualized.
+ * Numbering of the aforementioned and local vertice numbering of cells can be
+enabled by using enum flags.
+ * Combine the flags by using the binary or (|) operator to get a more detailed
+visualization of the mesh.\n
+ * Another option is to pass the corresponding integer value directly as an
+argument. See the enum definition for correct value.\n
+ *
+ * #### Output control flags:
+ * - TikzOutputCtrl::RenderCells to show the specific cells in the mesh in
+addition to the mesh grid
+ * - TikzOutputCtrl::CellNumbering to display numbering of cells
+ * - TikzOutputCtrl::VerticeNumbering to display local vertice numbering of
+cells
+ * - TikzOutputCtrl::NodeNumbering to display numbering of nodes
+ * - TikzOutputCtrl::EdgeNumbering to display edge numbering
+ *
+ *
+ * @note If no details about nodes, cells or edges are wanted, simply pass 0 as
+the 'int output_ctrl' parameter in writeTikZ(). This will draw only the mesh
+grid and nodes.
+ * @note Omitting the output_ctrl argument completely when calling the function:
+Cells, numbering of cells and numbering of vertices will be printed.
+(output_ctrl = 7)
+ * @note TikzOutputCtrl::RenderCells must be enabled in order to use the flags
+for numbering of cells and of local vertices of cells.
+ *
+ * In the LaTeX document, remember to include "\usepackage{tikz}". Use
+"\input{}" to include the code file and visualize the mesh.
+ *
+ * #### Examples of use
+ *
+ *
+ * ##### Function call
+ *
+ * \verbatim
+
+    // Enum flag for node numbering
+    writeTikZ(*mesh, "filename.txt", TikzOutputCtrl::NodeNumbering);
+
+    // Combining enum flags, enabling more detailed output
+    // The two examples are equivalent:
+    writeTikZ(*mesh, "filename.txt",
+TikzOutputCtrl::RenderCells|TikzOutputCtrl::EdgeNumbering|TikzOutputCtrl::VerticeNumbering);
+    writeTikZ(*mesh, "filename.txt", 21);
+    // Note that ::VerticeNumbering only works because ::RenderCells is
+activated.
+
+    // Without flags
+    writeTikZ(*mesh, "filename.txt",0);
+
+    // Without specifying last argument
+    writeTikZ(*mesh, "filename.txt"); is equivalent to writeTikZ(*mesh,
+"filename.txt", 7);
+
+ \endverbatim
+ *
+ * ##### LaTeX input
+ * \verbatim
+
+     \documentclass{article}
+     \usepackage{tikz}
+     \begin{document}
+
+     \input{"filename.txt"}
+
+     \end{document}
+\endverbatim
  *
  *
  */
+void writeTikZ(const lf::mesh::Mesh &mesh, const std::string &filename,
+               int output_ctrl = 7);
```

Comment:
I would suggest to change the type of the output_ctrl variable to TikzOutputCtrl and provide an operator| as it is suggested here: https://stackoverflow.com/a/1448478/2796908.

The big advantage of this is, that it is clear from the function signature what values are possible.

Reply:
How can you let | operator on two enums, since it may yield a value that is represented in the enumeration?

Reply:
From an implementation point of view it's not a problem, it works, since (scoped) enums are implemented as an integral type (by default int), https://softwareengineering.stackexchange.com/questions/194412/using-scoped-enums-for-bit-flags-in-c (I just realized that I provided a wrong link in my initial comment, fixed now)

From a conceptual point of view I understand that this may be cumbersome at first, however there are at least three languages that handle it exactly the same way for enums: C#, Visual Basic, Python, so for most programmers this is just normal. 

The big advantage ist, that `writeTikZ(mesh, "filename.tikz", TikzOutputControl::RenderCells | TikzOutputControl::CellNumbering)` is much more expressive than `writeTikz(mesh, "filename.tikz", 3)`.